src/standalone/intellisense/notebookPythonPathService.node.ts

```diff
@@ -44,6 +51,14 @@ export class NotebookPythonPathService implements IExtensionSingleActivationServ
             return;
         }
 
+        const pylanceApi = await this._getPylanceApi();
```

Comment:
This looks good and pretty straightforward to me. One question, is the python extension going to remove the registerJupyterPythonPathFunction and registerGetNotebookUriForTextDocumentUriFunction functions? I think that I mentioned this over Teams, but from the Jupyter perspective I don't see an issue with multiple APIs hooking up here. Say someone has a new Pylance, but an older Python, they might both register here. We could say that if we register pylance then don't register python. But from a Jupyter perspective that type of limitation doesn't seem needed. We just provide the functions for whoever wants them.

Reply:
for now, both of them will register and only one will be used depending on an option set. once we think everything is stable, it will be removed from python core extension (and any client related code).